### PR TITLE
Added ToDisplayString method to output a string for human consumption

### PR DESCRIPTION
**Pull request by preeder**
Recently needed to convey Duration to people programmatically and ToDisplayString does that.
Also added a .editorconfig file to ensure that authors follow a common style. You should check this; I was really only concerned with tabs settings.

### CODE REVIEW

File: src/Iso8601DurationHelper/Duration.cs [hunk 1]
```diff
@@ -158,8 +162,249 @@ void AppendComponent(uint number, string symbol)
...
+                    {
+                        value += Months / 12.0;
+                    }
+                    if (Weeks > 0)
+                    {
+                        value += Weeks / (365.0 / 7);
+                    }
+                    if (localDays > 0)
+                    {
+                        value += localDays / 365.0;
```
Comment (thomaslevesque) [inline src/Iso8601DurationHelper/Duration.cs:207]: Not all years have 365 days...

File: src/Iso8601DurationHelper/Duration.cs [hunk 2]
```diff
@@ -158,8 +162,249 @@ void AppendComponent(uint number, string symbol)
...
+                if (outputDecimalValue)
+                {
+                    double value = Months;
+                    if (Weeks > 0)
+                    {
+                        value += Weeks / (30.0 / 7);
+                    }
+                    if (localDays > 0)
+                    {
+                        value += localDays / 30.0;
```
Comment (thomaslevesque) [inline src/Iso8601DurationHelper/Duration.cs:240]: Not all months have 30 days

**Review by thomaslevesque — COMMENTED** [review]
Hi @preeder,

Thanks for your PR!

I understand the need to be able to display the duration as a human-readable form, but the problem is that it's always going to display inaccurate values, because not all years and months have the same number of days. ISO 8601 durations don't represent a constant duration; the actual duration in seconds can vary depending on the date. So you can't say that `P1Y1M` is 1.08 years, because it's not really true (if you wanted to render it as "1 year 1 month", for instance, that would make more sense, but I assume it's not what you want).

I created this library with the intent of avoiding this problem, which often comes up when you use `TimeSpan` to represent durations. So I'd rather not introduce such inaccuracies now.

So, while I appreciate your contribution, I can't accept it in its current form. If you really need it to work like this, I suggest you implement it as an extension method in your own code.

**Comment by preeder** [comment]
Ok, thanks for the quick response  and description.

**Pull request closed without merging**